### PR TITLE
engage-with-elksyncer

### PR DESCRIPTION
**Pull request by soyombo-baterdene**
[ISSUE](https://github.com/erxes/erxes/issues/ISSUE)

### Context

Your context here.  Additionally, any screenshots.  Delete this line.

// Delete the below section once completed
### PR Checklist
- [ ] Description is clearly stated under Context section
- [ ] Screenshots and the additional verifications are attached

### CODE REVIEW

File: api/src/__tests__/engageMessageDb.test.ts [hunk 1]
```diff
@@ -968,3 +1020,181 @@ describe('createVisitorOrCustomerMessages', () => {
...
+    });
+
+    _brand = await brandFactory({});
+    _integration = await integrationFactory({ brandId: _brand._id });
+    _user = await userFactory({});
+
+    const message = new EngageMessages({
+      title: 'Visitor',
+      fromUserId: _user._id,
+      kind: 'visitorAuto',
```
Comment (Buyantogtokh) [inline api/src/__tests__/engageMessageDb.test.ts:1082]: use constants

File: api/src/__tests__/engageMessageDb.test.ts [hunk 2]
```diff
@@ -968,3 +1022,181 @@ describe('createVisitorOrCustomerMessages', () => {
...
+
+    elkMock = sinon.stub(elk, 'fetchElk').callsFake(() => {
+      return Promise.resolve({
+        hits: {
+          hits: [
+            {
+              _id: faker.random.uuid(),
+              _source: {
+                kind: MESSAGE_KINDS.VISITOR_AUTO,
+                method: 'messenger',
```
Comment (Buyantogtokh) [inline api/src/__tests__/engageMessageDb.test.ts:1059]: use constant

File: api/src/db/models/Engages.ts [hunk 3]
```diff
@@ -386,13 +418,49 @@ export const loadClass = () => {
...
+
+      if (ELK_SYNCER === 'true') {
+        const must = [
+          { match: { 'engageData.messageId': engageData.messageId } },
+          { match: customerId ? { customerId } : { visitorId } }
+        ];
+
+        const response = await fetchElk(
+          'search',
+          'conversation_messages  ',
```
Comment (Buyantogtokh) [inline api/src/db/models/Engages.ts:433]: wtf is this empty space

**Review by batamar — COMMENTED** [review]
Engages iin model deer

createVisitorOrCustomerMessages function dotor bga Customers.findOne, Users.findOne 2 iig elk ashigladag bolgoh

**Pull request merged**